src/public/modules/forms/helpers/ndjsonStream.ts

```diff
@@ -0,0 +1,68 @@
+// Modified fork of https://github.com/canjs/can-ndjson-stream to enqueue
+// the string immediately without a JSON.parse() step, as the stream payload
+// is to be decrypted by the decryption worker.
+
+// Note that this code assumes a polyfill of TextDecoder is available to run in IE11.
+
+export const ndjsonStream = (
+  response: ReadableStream<Uint8Array>,
+): ReadableStream => {
+  // For cancellation
+  let maybeReader: ReadableStreamDefaultReader<Uint8Array> | undefined
+  let shouldCancel = false
+  return new ReadableStream({
+    start: function (controller) {
+      const reader = response.getReader()
+      maybeReader = reader
+      const decoder = new TextDecoder()
+      let data_buf = ''
+
+      return reader
+        .read()
+        .then(function processResult(result): Promise<void> | undefined | void {
+          if (shouldCancel) {
+            return
+          }
```

Comment:
noticed you've shifted this block outside the `if (result.done)` - why?

Reply:
i just found it really hard to read with the nested if conditions @__@ i think it's clearer this way but do feel free to disagree!

Reply:
but the behaviour is different right - previously we would return early only if `result.done && shouldCancel`, but now you're returning early if `shouldCancel` is true, regardless of `result.done`?

Reply:
oh shit you're absolutely right. great catch and i'll edit the initial condition!